monkestation/code/modules/virology/machines/curer.dm

```diff
@@ -93,7 +93,8 @@
 	var/datum/reagent/blood/B = locate() in container.reagents.reagent_list
 
 	var/list/data = list()
-	data["antigen"] = B.data["immunity"]
-
-	product.reagents.add_reagent(/datum/reagent/vaccine , 30, data)
+	var/list/immunity = B.data["immunity"]
+	if(istype(immunity) && immunity.len > 0)
```

Comment:
```suggestion
	if(length(immunity))
```
`list.len` should never be used over `length(list)` unless it is actually needed (it is not needed for simple comparisons / empty checks)